### ISSUE TITLE
ContentSets repo file not being copied into image w/o --redhat

### ISSUE THREAD

**Issue by jmtd** [cekit/cekit#491]
**Describe the bug**
When building an image that uses content_sets with cekit 3.0.0 on Fedora 29 wtih the podman back end, the .repo file for the ContentSets is not getting copied into the image, **unless** you also specify `--redhat` to `cekit`.

**To reproduce**
This is [this image](https://github.com/jboss-container-images/openjdk/commit/b8f0c0db0e08ef137b984f1aac411555a5a54bc0), develop branch, commit b8f0c0db0e08ef137b984f1aac411555a5a54bc0

```
(3.0.0) [jdowland@brick openjdk]$ cekit build --overrides-file rhel7-jdk8-overrides.yaml podman
2019-04-23 14:59:00,807 cekit        INFO     You are running on known platform: Fedora 29 (Workstation Edition)
2019-04-23 14:59:00,875 cekit        INFO     Generating files for podman engine
2019-04-23 14:59:00,880 cekit        INFO     Initializing image descriptor...
2019-04-23 14:59:00,892 cekit        INFO     Preparing resource 'cct_module'
2019-04-23 14:59:04,007 cekit        WARNING  The image has ContentSets repositories specified, all other repositories are removed!
2019-04-23 14:59:04,013 cekit        INFO     Handling artifacts...
2019-04-23 14:59:04,013 cekit        INFO     Preparing resource 'hawkular-javaagent'
2019-04-23 14:59:04,032 cekit        INFO     Preparing resource 'jolokia-jvm-1.5.0.redhat-1-agent.jar'
2019-04-23 14:59:04,033 cekit        INFO     Preparing resource 'jmx_prometheus_javaagent'
2019-04-23 14:59:04,034 cekit        INFO     Rendering Dockerfile...
2019-04-23 14:59:04,073 cekit        INFO     Building container image...
STEP 1: FROM rhel7:7-released
STEP 2: USER root
--> Using cache 2b7e2e9722ed00fa4d9e8a0d11881dbfe1dd0c9823f963a45075ca3da6967bf4
STEP 3: FROM 2b7e2e9722ed00fa4d9e8a0d11881dbfe1dd0c9823f963a45075ca3da6967bf4
STEP 4: RUN yum makecache
Loaded plugins: ovl, product-id, search-disabled-repos, subscription-manager
This system is not registered with an entitlement server. You can use subscription-manager to register.
There are no enabled repos.
 Run "yum repolist all" to see the repos you have.
 To enable Red Hat Subscription Management repositories:
     subscription-manager repos --enable <repo>
 To enable custom repositories:
     yum-config-manager --enable <repo>
Error: error building at step {Env:[PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin container=oci] Command:run Args:[yum makecache] Flags:[] Attrs:map[] Message:RUN yum makecache Original:RUN yum makecache}: error while running runtime: exit status 1
2019-04-23 14:59:05,731 cekit        ERROR    Image build failed, see logs above.
(3.0.0) [jdowland@brick openjdk]$ podman run --rm -ti 2b7e2e9722ed00fa4d9e8a0d11881dbfe1dd0c9823f963a45075ca3da6967bf4 ls /etc/yum.repos.d
redhat.repo
```

**Expected behavior**
There should be an additional .repo file in /etc/yum.repos.d corresponding to the ODCS content sets, OR a fatal error that content sets are not supported w/o `--redhat` being specified.

**Comment by goldmann** [comment]
I can see two problems here:

1. You are trying to build a RHEL image without the `--redhat` switch which is just a mistake.
2. The tool says that content sets are used, but it does not even try to apply them in case of local builders
3. ODCS support is missing in local builders other than Docker

What the tool should do is to request an ODCS compose using the community ODCS service, which is running in the Fedora infrastructure. This would eventually **result in a failure**, because the content sets defined in your case won't be found there. Which is perfectly fine, and expected.

The solution for this issue would be to request ODCS compose and fail in case the compose cannot be created. Log message should be added which ODCS server is used to make easier to spot problems like the missing `--redhat` switch.

**Comment by goldmann** [comment]
A proposed fix is available in #497.
